GSA/COM API/Python/GSA2Compos.py

```diff
@@ -0,0 +1,202 @@
+"""
+Program for linking GSA with Compos for composite beam design
+This script is intended to
+1. open GSA model using gsapy
+2. extract member sections and lengths based on first saved list
+   - write to member_list
+3. extract defined profiles and write to sections_dictionary
+4. write compos_file.csv file using the member_list
+5. open compos_file.csv in Compos, design the members &
+   extract the new sections to compos_results
+6. reopen the GSA model,
+   use compos_results and sections_dictionary to update the members
+"""
+
+import win32com.client  # https://pypi.org/project/pywin32/
+from datetime import datetime
+from tempfile import gettempdir
+from gsapy import GSA
+
+# gsapy references
+# https://gitlab.arup.com/david.dekoning/gsapy/-/blob/master/docs/readme.rst
+# https://gitlab.arup.com/hongkong-rd/JupyterSamples/blob/master/pyGSA/pyGSA_gsapy.ipynb
+
+
+dead_load = 3.0  # kN/m^2
+live_load = 5.0  # kN/m^2
+# add additional variables as necessary to control what is added to Compos
+
+
+def gsa2compos2gsa():
+    global dead_load
+    global live_load
+
+    # get GSA data
+    print("Opening GSA model")
+    gsa_model = GSA(r'C:\Temp\Composite test.gwb', version='10.1')
+
+    # gets the members from the first saved list
+    search_list = \
+        str(gsa_model.get_all_saved_lists()[0]).split("\t")[4].split(" ")
+    member_list = []
+    for i in search_list:  # create list of members to be designed
+        member = str(gsa_model.get_members(int(i)))
+        member_section = int(member.split("\t")[6])
+        member_list.append([int(i), gsa_model.get_section_info(member_section),
+                            gsa_model.get_member_length(int(i))])
+    sections_dictionary = {}
+    for i in range(1, 93):  # create sections dictionary:  needs to match model
+        sections_dictionary[gsa_model.get_section_info(i)] = i
+    gsa_model.close()
+    print("  GSA model closed")
+
+    # create Compos file
+    print("Creating Compos file")
+    now = datetime.now().strftime("%A, %d %b %Y at %H:%m:%S")
+    # creates Compos file in system temp directory
+    with open(gettempdir() + '\\compos_file.csv', 'w') as compos_file:
+        compos_file.write("! This file was originally written by GSA2Compos.py"
+                          " on " + now)
+        compos_file.write("\nCOMPOS_FILE_VERSION,1")
+        compos_file.write("\nTITLE,,,,,")
+        compos_file.write("\nUNIT_DATA,FORCE,kN,0.00100000")
+        compos_file.write("\nUNIT_DATA,LENGTH,m,1.00000")
+        compos_file.write("\nUNIT_DATA,DISP,mm,1000.00")
+        compos_file.write("\nUNIT_DATA,SECTION,cm,100.000")
+        compos_file.write("\nUNIT_DATA,STRESS,N/mm²,1.00000e-006")
+        compos_file.write("\nUNIT_DATA,MASS,t,0.00100000")
+        for i in range(len(member_list)):
```

Comment:
the more python is way is to iterate over the list, as opposed to incrementing a counter:

```python
for beam_number, beam_section, beam_length in member_list:
    compos_file.write ...
```

since member_list is a list of tuples, you can expand the tuple in the for loop